tiledb/sm/consolidator/fragment_consolidator.cc

```diff
@@ -558,47 +558,32 @@ Status FragmentConsolidator::consolidate_internal(
   cw.resize_buffers(stats_, config_, array_schema, average_var_cell_sizes);
 
   // Create queries
-  auto query_r = (Query*)nullptr;
-  auto query_w = (Query*)nullptr;
-  auto st = create_queries(
+  tdb_unique_ptr<Query> query_r = nullptr;
+  tdb_unique_ptr<Query> query_w = nullptr;
+  RETURN_NOT_OK(create_queries(
       array_for_reads,
       array_for_writes,
       union_non_empty_domains,
-      &query_r,
-      &query_w,
-      new_fragment_uri);
-  if (!st.ok()) {
```

Comment:
```
if(!st.ok()) {
  throw_if_not_ok(st);
  return st; // I think we need that for windows
}
```
seems closer than the current guideline of always trowing.,

Reply:
Changed `RETURN_NOT_OK` to `throw_if_not_ok`; it should be the same thing as your proposal.